class-tgm-plugin-activation.php

```diff
@@ -1128,6 +1108,75 @@ public function _get_plugin_data_from_name( $name, $data = 'slug' ) {
 		}
 
 		/**
+		 * Retrieve the download url for a package.
+		 *
+		 * @since 2.5.0
+		 *
+		 * @param string $slug
+		 *
+		 * @return string Plugin download URL or path to local file
+		 */
+		public function get_download_url( $slug ) {
+			$dl = '';
+
+			if ( ! empty( $this->plugins[ $slug ]['source'] ) ) {
```

Comment:
I prefer to see big conditions like this returned early as a guard clause.

~~~php
if ( empty( $this->plugins[ $slug ]['source'] ) ) {
    return '';
}
~~~

Use of `''` makes it clearer that we're intentionally returning an empty string, and not a valid URL.